src/github_utils.py

```diff
@@ -19,48 +19,56 @@ def get_pull_request_diff():
     # Get diff content
     diff_url = pull_request.diff_url
     response = requests.get(diff_url)
-    print(response.text)
+    #print(response.text)
     return response.text
 
 
-def get_valid_lines(diff_content):
-    """Parse diff to find valid line numbers with proper initialization"""
-    valid_lines = {}
+def get_valid_hunks(diff_content):
+    """Parse diff to find valid hunk positions"""
+    hunks = []
+    current_hunk = None
     current_file = None
-    line_number = None
     
     for line in diff_content.split('\n'):
-        # Reset tracking for new files
         if line.startswith('diff --git'):
-            current_file = line.split(' b/')[-1].strip()
-            valid_lines[current_file] = set()
-            line_number = None  # Reset for new file
-            continue
+            # New file section
+            current_file = line.split(' b/')[1].split()[0]
+        elif line.startswith('@@'):
+            # Hunk header format: @@ -old_start,old_lines +new_start,new_lines @@
+            parts = line.split(' ')
+            new_part = parts[2].split(',')
+            new_start = int(new_part[0][1:])
+            hunk_length = int(new_part[1])
```

Comment:
Consider using a more descriptive variable name than `parts`. Also, add error handling or validation for the structure of `parts` to avoid potential `IndexError` issues if the line format is unexpected.

```suggestion
parts = line.split(' ')
```
```suggestion
hunk_parts = line.split(' ')
```if len(hunk_parts) < 3:
print(f"Invalid hunk header format: {line}")
continue